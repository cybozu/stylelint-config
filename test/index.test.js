import stylelint from "stylelint";
import fs from "node:fs";
import path from "node:path";
import url from "node:url";
import { it, expect } from "@jest/globals";
import styledComponentsConfig from "../styled-components";
import config from "../index";

const __dirname = path.dirname(url.fileURLToPath(import.meta.url));
const getSnapshot = (result) =>
  result.results.map(({ warnings }) =>
    warnings.map(({ text, ...rest }) => rest),
  );
const getCode = (file) =>
  fs.readFileSync(path.resolve(__dirname, "fixtures", file), "utf-8");

it("should not raise any errors with valid.css", async () => {
  const result = await stylelint.lint({
    code: getCode("valid.css"),
    config,
  });
  expect(result.errored).toBeFalsy();
});

it("should raise errors with invalid.css", async () => {
  const result = await stylelint.lint({
    code: getCode("invalid.css"),
    config,
  });
  expect(result.errored).toBeTruthy();
  expect(getSnapshot(result)).toMatchSnapshot();
});

it("should verify CSS in styled-components", async () => {
  const result = await stylelint.lint({
    code: getCode("Styled.tsx"),
    config: styledComponentsConfig,
  });
  expect(result.errored).toBeTruthy();
  expect(getSnapshot(result)).toMatchSnapshot();
});
