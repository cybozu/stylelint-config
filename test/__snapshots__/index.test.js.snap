// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should raise errors with invalid.css 1`] = `
[
  [
    {
      "column": 24,
      "endColumn": 26,
      "endLine": 6,
      "line": 6,
      "rule": "alpha-value-notation",
      "severity": "error",
    },
    {
      "column": 10,
      "endColumn": 27,
      "endLine": 6,
      "line": 6,
      "rule": "color-function-notation",
      "severity": "error",
    },
    {
      "column": 3,
      "endColumn": 17,
      "endLine": 2,
      "line": 2,
      "rule": "custom-property-pattern",
      "severity": "error",
    },
    {
      "column": 16,
      "endColumn": 18,
      "endLine": 10,
      "line": 10,
      "rule": "length-zero-no-unit",
      "severity": "error",
    },
    {
      "column": 24,
      "endColumn": 26,
      "endLine": 10,
      "line": 10,
      "rule": "length-zero-no-unit",
      "severity": "error",
    },
    {
      "column": 1,
      "endColumn": 5,
      "endLine": 9,
      "line": 9,
      "rule": "no-descending-specificity",
      "severity": "error",
    },
    {
      "column": 20,
      "endColumn": 21,
      "endLine": 6,
      "line": 6,
      "rule": "no-irregular-whitespace",
      "severity": "error",
    },
  ],
]
`;

exports[`should verify CSS in styled-components 1`] = `
[
  [
    {
      "column": 30,
      "endColumn": 2,
      "endLine": 21,
      "line": 6,
      "rule": "no-missing-end-of-source-newline",
      "severity": "error",
    },
    {
      "column": 24,
      "endColumn": 26,
      "endLine": 10,
      "line": 10,
      "rule": "alpha-value-notation",
      "severity": "error",
    },
    {
      "column": 10,
      "endColumn": 27,
      "endLine": 10,
      "line": 10,
      "rule": "color-function-notation",
      "severity": "error",
    },
    {
      "column": 16,
      "endColumn": 18,
      "endLine": 19,
      "line": 19,
      "rule": "length-zero-no-unit",
      "severity": "error",
    },
    {
      "column": 24,
      "endColumn": 26,
      "endLine": 19,
      "line": 19,
      "rule": "length-zero-no-unit",
      "severity": "error",
    },
    {
      "column": 1,
      "endColumn": 5,
      "endLine": 18,
      "line": 18,
      "rule": "no-descending-specificity",
      "severity": "error",
    },
    {
      "column": 20,
      "endColumn": 21,
      "endLine": 10,
      "line": 10,
      "rule": "no-irregular-whitespace",
      "severity": "error",
    },
  ],
]
`;
